### PR TITLE
refactor(core): Stop reporting non-error to sentry (issue 4229454473) (no-changelog)

### PR DESCRIPTION
**Pull request by krynble**
This PR aims to stop reporting issues such as [this one](https://n8nio.sentry.io/issues/4229454473/events/42b96bfd6a334c15a84499e981cf90eb/?project=4503924908883968).

Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/cli/src/ActiveExecutions.ts [hunk 1]
```diff
@@ -190,13 +190,17 @@ export class ActiveExecutions {
...
-		// Create the promise which will be resolved when the execution finished
-		const waitPromise = await createDeferredPromise<IRun | undefined>();
-
 		if (this.activeExecutions[executionId] === undefined) {
-			throw new Error(`There is no active execution with id "${executionId}".`);
+			const toThrow = new WorkflowOperationError(
+				`There is no active execution with id "${executionId}".`,
+			);
+			toThrow.severity = 'warning';
+			throw toThrow;
```
Comment (netroy) [inline packages/cli/src/ActiveExecutions.ts:198]: looking at `WorkflowOperationError` and `SubworkflowOperationError`, I'm not sure if there are any cases so far where the `severity` should not be `warning`.
Maybe we should update those two constructors to include `this.severity = 'warning';`, and replace this with
```suggestion
			throw new WorkflowOperationError(
				`There is no active execution with id "${executionId}".`,
			);
```

**Review by netroy — CHANGES_REQUESTED** [review]

File: packages/cli/src/ActiveExecutions.ts (hunk 1 above)
Comment (krynble) [reply packages/cli/src/ActiveExecutions.ts]: Good point, changed!

File: packages/workflow/src/WorkflowErrors.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export class WorkflowOperationError extends ExecutionBaseError {
 
 	constructor(message: string, node?: INode) {
 		super(message, { cause: undefined });
+		this.severity = 'warning';
```
Comment (ivov) [inline packages/workflow/src/WorkflowErrors.ts:18]: To make sure, we want to stop reporting all `WorkflowOperationError` and `SubworkflowOperationError`? As we would lose e.g. errors like [this](https://github.com/n8n-io/n8n/blob/1055bd3762b90b013a300bd87e3fa902e902cb9e/packages/cli/src/eventbus/MessageEventBus/recoverEvents.ts#L88-L90).

Comment (netroy) [reply packages/workflow/src/WorkflowErrors.ts:18]: I don't think we want to report that Sentry. We should use Sentry for for tracking unhandled and actionable errors. 
If we want to keep a track of possible OOM crashes, then we should use logging or telemetry for that.

Comment (netroy) [reply packages/workflow/src/WorkflowErrors.ts:18]: I looked into all instance of `WorkflowOperationError` and `SubworkflowOperationError`, and they all looked like warnings to me. But if you think some of them should be reported, then we should update the constructors to accept `severity` in the `options` argument, just like how `NodeOperationError` does.

File: packages/workflow/src/WorkflowErrors.ts [hunk 3]
```diff
@@ -28,6 +29,7 @@ export class SubworkflowOperationError extends WorkflowOperationError {
 
 	constructor(message: string, description: string) {
 		super(message);
+		this.severity = 'warning';
```
Comment (netroy) [inline packages/workflow/src/WorkflowErrors.ts:32]: since `SubworkflowOperationError` extends `WorkflowOperationError`, this line is redundant.

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.15.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.15.1)

**Pull request merged**